src/hooks/useSemester.tsx

```diff
@@ -0,0 +1,22 @@
+import { useQuery, useMutation } from 'react-query';
+import { Semester } from 'api';
+import { AddSemesterData } from 'components/AddSemesterModal';
+
+import {
+	fetchGetSemester,
+	fetchPostSemester,
+	fetchPutSemester,
+	fetchDeleteSemester,
```

Comment:
if the above names `getSemester` and friends are too similar to the names in the function body, we can always use the `as` keyword to rename locally.